third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/closures/bug_16406.move

```diff
@@ -0,0 +1,49 @@
+//# publish
+module 0xc0ffee::m {
+
+    inline fun id(f: ||u64 has copy+drop): ||u64 {
+        f
+    }
+
+    inline fun id_2(f: ||u64 has copy+drop): ||u64 {
+        let k = f;
+        let y = k();
+        || y + f()
+    }
+
+    inline fun id_3(f1: ||u64 has copy+drop, x: u64, f2: ||u64 has copy+drop): ||u64 {
+        let k = f1;
+        let y = k();
+        || y + f2() + x
+    }
+
+    inline fun id_4(f1: ||u64 has copy+drop, x: u64, f2: ||u64 has copy+drop): ||u64 {
+        let k = f2;
```

Comment:
I think we are adding new functionality here, and I am not sure if that is the right thing to do (expanding what inline functions can do).

In particular, previously, the only thing we could do with inline function's function parameters is apply them (or not use them at all). Here, it looks like it can be assigned to other variables, returned, etc.

Reply:
cc @wrwg on whether we should allow a function parameter to be used as a normal local in the body of an inline function.

Reply:
We should probably not allow this, because a recent PR made the type checking rules for those different.  So you would get a bytecode verification error in some cases. Lets better create an error message instead, if we don't have one yet.

Reply:
Updated the code to generate the error instead of allowing it.